controllers/front/submitplus.php

```diff
@@ -0,0 +1,266 @@
+            $order    = new Order((int) $this->id_order);
+            $currency = new Currency((int) $order->id_currency);
+
+            if (Validate::isLoadedObject($order)) {
+                $params                 = array();
+                $params['objOrder']     = $order;
+                $params['currencyObj']  = $currency;
+                $params['currency']     = $currency->sign;
+                $params['total_to_pay'] = $order->getOrdersTotalPaid();
+
+                return $params;
+            }
+        }
+
+        return false;
+    }
+
+    public function displayAjax()
+    {
+        $ajax = Tools::getValue('ajax');
+        if (!$ajax) {
+            $return['error'][] = 'Une erreur est survenue pendant le paiement.';
+            echo json_encode($return);
+            die();
+        }
+
+        $id_cart   = Tools::getValue('id_cart');
+        $payerID   = Tools::getValue('payerID');
+        $paymentId = Tools::getValue('paymentId');
+        $submit    = Tools::getValue('submit');
+
+        if (
+            (!empty($id_cart) && $this->context->cart->id == $id_cart) &&
+            !empty($payerID) &&
+            !empty($paymentId) &&
+            !empty($submit)
+        ) {
+
+            $CallApiPaypalPlus = new CallApiPaypalPlus();
+            $payment           = json_decode($CallApiPaypalPlus->executePayment($payerID, $paymentId));
+
+            if (isset($payment->state)) {
+
+                $paypal = new PayPal();
+
+                $transaction = array(
+                    'id_transaction' => $payment->transactions[0]->related_resources[0]->sale->id,
+                    'payment_status' => $payment->state,
+                    'total_paid' => $payment->transactions[0]->amount->total,
+                    'id_invoice' => 0,
+                    'shipping' => 0,
+                    'currency' => $payment->transactions[0]->amount->currency,
+                    'payment_date' => date("Y-m-d H:i:s"),
+                );
+
+                if ($submit == 'confirmPayment') {
+
+                    if ($payment->state == 'approved') {
+
+                        $paypal->validateOrder(
+                            $this->id_cart, $this->getOrderStatus('payment'), $payment->transactions[0]->amount->total,
+                            $payment->payer->payment_method, null, $transaction
+                        );
+                        $return['success'][] = 'Votre paiement a bien été pris en compte.';
+                    } else {
+
+                        $paypal->validateOrder(
+                            $this->id_cart, $this->getOrderStatus('payment_error'),
+                            $payment->transactions[0]->amount->total, $payment->payer->payment_method, null,
+                            $transaction
+                        );
+                        $return['error'][] = 'Une erreur est survenue pendant le paiement.';
```

Comment:
Displayed text should be written first in English and then translated.

You have already everything you need for that with `$paypal->l(...)`